### ISSUE TITLE
Proposal: Introduce raw manifest resource

### ISSUE THREAD

**Issue by burdiyan** [terraform-providers/terraform-provider-kubernetes#109]
It would be cool if this provider would support creating any resource valid for the current api server just directly in YAML. So that resources that are not natively supported by the provider can still be created using YAML (the fact of having <<HEREDOC for HCL properties is super useful for this) using the behavior similar to `kubectl apply -f`.

**Comment by ericchiang** [comment]
I've been working on a project that does exactly this. https://github.com/ericchiang/terraform-provider-k8s

Would love feedback.

**Comment by burdiyan** [comment]
Cool! Will take a look at it soon. I believe it still would be great to have that in the official Kubernetes provider. 

Have you thought about contributing that back to this provider @ericchiang?

**Comment by ericchiang** [comment]
> Have you thought about contributing that back to this provider @ericchiang?

I didn't plan to. Would be happy to review PRs from anyone who wanted to add the feature to this one.

**Comment by radeksimko** [comment]
Hi folks 👋 
thanks for raising this question.

Supporting YAML/JSON was considered before and during the initial implementation of this provider and we decided _not_ to do it. For that reason I'm going to close this issue as we have no intentions of implementing it nor accepting PRs which accept raw configs in JSON/YAML.

Feel free to use @ericchiang's provider if you wish.

Full discussion: https://github.com/hashicorp/terraform/pull/3132

TL;DR version (summary): https://github.com/hashicorp/terraform/pull/3453#issuecomment-189733008
